kafka-connect-cassandra/src/main/scala/com/datamountaineer/streamreactor/connect/cassandra/source/CassandraTableReader.scala

```diff
@@ -340,9 +344,6 @@ class CassandraTableReader(private val name: String,
     while (!queue.offer(record, 1, TimeUnit.SECONDS) & !shouldStop) {
       shouldStop = stop.get()
     }
-    if(shouldStop){
-      throw new ConnectException(s"Connector $name stopped for table $keySpace.$table.")
```

Comment:
@stheppi,if remove this, a connector can wait 100 seconds for stop. It is for force stop. This exception is catched in [this line](https://github.com/Landoop/stream-reactor/blob/92987296ae78774befa6bc56e6c12b71de6b2a3a/kafka-connect-cassandra/src/main/scala/com/datamountaineer/streamreactor/connect/cassandra/source/CassandraTableReader.scala#L249)

Reply:
i pushed a change.
the offer was called first before the check on the stop

now that is avoid. previousely the loop was stop by an exception. not an ideal way breaking the loop

Reply:
About valatile cache. Don't have problem remove?

Reply:
it will affect the code a bit in terms of performance. but we check the variable in the processRow anyway. so not much reason to cache it then